### DISCUSSION TITLE
Thoughts about issue markdown sanitization

### DISCUSSION THREAD

**Discussion by jose-ws** [renovatebot/renovate#10471] [Suggest an Idea]
In regards to issue #8030

1.  Currently most of the platforms, when they call ensureIssue(), are calling massageMarkdown() to preprocess the raw body of the issue. The massageMarkdown() function seems like a good place to call sanitizeMarkdown() before it returns the modified markdown. sanitizeMarkdown() contains the fix for #8030. If this is ok, I can make a PR with the change.

[https://github.com/renovatebot/renovate/blob/5b25bf5e0c08c239439f02edcef8788184581a60/lib/platform/gitlab/index.ts#L798](https://github.com/renovatebot/renovate/blob/5b25bf5e0c08c239439f02edcef8788184581a60/lib/platform/gitlab/index.ts#L798)

2. for the github platform there is a massageMarkdown() function that exists with some business logic but it is not being used. Is this a bug? If so I can bring it back in to ensureIssue() and potentially also call sanitizeMarkdown() in massageMarkdown().

[https://github.com/renovatebot/renovate/blob/5c21d440555e9bb236ede506e7627e2fa1bd5f4c/lib/platform/github/index.ts#L1653-L1663](https://github.com/renovatebot/renovate/blob/5c21d440555e9bb236ede506e7627e2fa1bd5f4c/lib/platform/github/index.ts#L1653-L1663)

**Comment by rarkins** [comment]
1. You're proposing to call the sanitize function from with each platform's `massageMarkdown()`?

2. `platform.massageMarkdown()` is called from multiple places, so if the platform is `github` then it will be called:
![image](https://user-images.githubusercontent.com/6311784/122281649-c8b66d80-ceea-11eb-85ee-f7c9dde0448a.png)

**Reply by jose-ws** [reply]
1. Yes
2. You didn't address this. Why is massageMarkdown being called from ensureIssue() for gitlab and bitbucket, but not github?

Let me ask it differently:
If platform calls to massageMarkdown handle all the markdown cleanup, then why is it called in gitlab.ensureIssue() and bitbucket.ensureIssue()?
If platform calls to massageMarkdown don't handle all the markdown cleanup then shouldn't it also be called from github.ensureIssue()?

**Reply by rarkins** [reply]
Let me re-answer (2) again from your original question first. You wrote:

> for the github platform there is a massageMarkdown() function that exists with some business logic

Correct

> but it is not being used. Is this a bug?

Incorrect because it *is* being used even if not directly by the github file itself, so not directly a bug either. `platform.massageMarkdown()` is called from lib/workers multiple times.

In the follow-up:

> Why is massageMarkdown being called from ensureIssue() for gitlab and bitbucket, but not github?

This is a good question and you've found an inconsistency. I don't mind if you add in `massageMarkdown()` to the platforms not utilizing it. Then you would add `sanitizeMarkdown()` to each platform's `massageMarkdown()` ?
